### PR TITLE
Atomic-free JNI work

### PR DESCRIPTION
**Pull request by gacholio**
- ensure vm state at every init stage
- optimize JNI critical enter and exit

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

File: runtime/vm/jvminit.c [hunk 1]
```diff
@@ -2960,6 +2961,22 @@ processVMArgsFromFirstToLast(J9JavaVM * vm)
...
+#if defined(J9VM_INTERP_ATOMIC_FREE_JNI)
+		if (mainThread->inNative) {
+			enterVMFromJNI(mainThread);
+			releaseVMAccess(mainThread);
+		} else
+#endif /* J9VM_INTERP_ATOMIC_FREE_JNI */
+		if (J9_ARE_ANY_BITS_SET(mainThread->publicFlags, J9_PUBLIC_FLAGS_VM_ACCESS)) {
+			releaseVMAccess(mainThread);
+		}
+	}
```
Comment (DanHeidinga) [inline runtime/vm/jvminit.c]: Does this have any effect on startup performance?

Comment (gacholio) [reply runtime/vm/jvminit.c]: It runs about 10 times, so I can't imagine it does.  I suspect it's mostly no-op anyway, if the stage leaves itself in the correct state on exit.

Comment (DanHeidinga) [reply runtime/vm/jvminit.c]: As far as we're aware, there was only a single stage that didn't obey this rule.  Sounds like it'll be pretty close to a nop

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux jdk8

**Pull request merged**